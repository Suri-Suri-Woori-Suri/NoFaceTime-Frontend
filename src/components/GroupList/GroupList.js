import React, { useState, useEffect } from 'react';
import styles from './GroupList.module.css';
import { getMember } from '../../api';

const GroupList = ({ groups, checkedGroup, setCheckedGroup, setgroupId, setShowMember, setExistMember }) => {
  const [isFetched, setIsFetched] = useState(false);

  //toggle 변수명 수정??
  const toggle = (groupId) => {
    console.log('Get Member Toggle');
    setgroupId(groupId);
    setIsFetched(!isFetched);
  };

  useEffect(() => {
    if (!isFetched) return;
    console.log('Get Member!!!!');
    //getMember(groupId);

    const response = [1, 2, 3, 4, 5, 6, 6, 1, 2, 3, 4, 5, 6];
    setExistMember([...response]);
    setIsFetched(false);
    setShowMember(true);
  }, [isFetched]);

  const onChange = groupId => {
    const clickedGroup = checkedGroup.indexOf(groupId);
    const currentCheckedGroup = [...checkedGroup];

    if (clickedGroup === -1) {
      currentCheckedGroup.push(groupId);
    } else {
      currentCheckedGroup.splice(clickedGroup, 1);
    }

    setCheckedGroup([...currentCheckedGroup]);
  };

  const groupList = groups ? groups.map((group, i) => {
    return (
      <div key={i}>
        <input
          type='checkbox'
          onChange={() => onChange(group._id)}
        />
        <button key={i} className={styles.Group} onClick={() => toggle(group._id)}>
          {group.name}
        </button>
      </div>

    );
  }) : undefined;

return <div className={styles.Groups}>{groupList}</div>
};

export default GroupList;
