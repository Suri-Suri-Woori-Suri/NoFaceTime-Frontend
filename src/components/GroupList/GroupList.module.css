.GroupList {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 90%;
  height: 60%;
  border: 10px solid yellow;
  box-sizing: border-box;
}

.GroupEntry {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  width: 90%;
  height: 10%;
  overflow: auto;
  border: 1px solid #c9d3dd;
  border-radius: 10px;
  background: rgb(255, 245, 188);
}

.GroupEntry:hover {
  background: rgb(243, 213, 43);
}
