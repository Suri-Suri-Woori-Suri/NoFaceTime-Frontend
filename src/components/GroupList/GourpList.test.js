import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import GroupList from './GroupList';

describe('<GroupList/>', () => {
  afterEach(cleanup);

  const mockGroups = [
    {
      name: 'group1',
      _id: 'a',
    },
    {
      name: 'group2',
      _id: 'b',
    }
  ];

  const mockGroupItemOnClick = jest.fn();
  const mockGroupNameOnClick = jest.fn();
  const mockCheckBox = jest.fn();

  const wrapper = mount(
    <GroupList
      groups={mockGroups}
      onGroupItemClick={mockGroupItemOnClick}
      onGroupNameClick={mockGroupNameOnClick}
      handleCheckbox={mockCheckBox}
    />
  );

  it('should be render lists correctly', () => {
    expect(wrapper.containsMatchingElement(mockGroups[0].name)).toEqual(true);
    expect(wrapper.containsMatchingElement(mockGroups[1].name)).toEqual(true);
  });

  it('should be invoked when button is clicked', () => {
    render(
      <GroupList
        groups={mockGroups}
        onGroupItemClick={mockGroupItemOnClick}
        onGroupNameClick={mockGroupNameOnClick}
        handleCheckbox={mockCheckBox}
      />
    );

    fireEvent.click(screen.getByText(mockGroups[0].name));

    expect(mockGroupItemOnClick).toHaveBeenCalledTimes(1);
    expect(mockGroupNameOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
