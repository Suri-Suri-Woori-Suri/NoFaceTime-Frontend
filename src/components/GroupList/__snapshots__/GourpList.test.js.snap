// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupList/> matches snapshot 1`] = `
<GroupList
  groups={
    Array [
      Object {
        "_id": "a",
        "name": "group1",
      },
      Object {
        "_id": "b",
        "name": "group2",
      },
    ]
  }
  handleCheckbox={[MockFunction]}
  onGroupItemClick={[MockFunction]}
  onGroupNameClick={[MockFunction]}
>
  <div
    className="GroupList"
  >
    <div
      className="GroupEntry"
      key="0"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="GroupName"
        key="0"
        onClick={[Function]}
      >
        group1
      </div>
    </div>
    <div
      className="GroupEntry"
      key="1"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="GroupName"
        key="1"
        onClick={[Function]}
      >
        group2
      </div>
    </div>
  </div>
</GroupList>
`;
